src/Bedrock.Framework/Protocols/WebSockets/WebSocketFrameReader.cs

```diff
@@ -21,17 +28,59 @@ public struct WebSocketFrameReader : IMessageReader<WebSocketReadFrame>
         /// <returns>True if parsed successfully, false otherwise.</returns>
         public bool TryParseMessage(in ReadOnlySequence<byte> input, ref SequencePosition consumed, ref SequencePosition examined, out WebSocketReadFrame message)
         {
-            var reader = new SequenceReader<byte>(input);
-
             //We need to at least be able to read the start of frame header
             if (input.Length < 2)
             {
                 message = default;
                 return false;
             }
 
-            reader.TryRead(out var finOpcodeByte);
-            reader.TryRead(out var maskLengthByte);
+            if (input.IsSingleSegment || input.FirstSpan.Length >= 14)
```

Comment:
We need to encapsulate this logic in a helper (it's impossible with stackalloc though).

Reply:
I'm not following you, what's a helper in this context and why do we need to?

Reply:
A method in the framework that can get you a span of bytes given a minimum length.

Reply:
Ah, I gotcha. Yeah, it's hard to picture a way of implementing that without having to either go to the pool or allocate a byte array, because you can't leak the stackalloc'd span.